src/opossum_lib/cli.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2023 TNG Technology Consulting GmbH <https://www.tngtech.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging
+import sys
+from pathlib import Path
+
+import click
+from spdx.model.document import Document
+from spdx.parser.error import SPDXParsingError
+from spdx.parser.parse_anything import parse_file
+
+from opossum_lib.file_generation import generate_json_file_from_tree, write_dict_to_file
+from opossum_lib.graph_generation import generate_graph_from_spdx
+from opossum_lib.tree_generation import generate_tree_from_graph
+
+
+@click.command()
+@click.option(
+    "--infile",
+    "-i",
+    help="The file containing the document to be converted.",
+)
+@click.option(
+    "--outfile",
+    "-o",
+    help="The file path to write the generated opossum document to. The generated file "
+    "will be in JSON format, if the specified file path doesn't match this file "
+    'extension ".json" will be appended.',
+)
+def main(infile: str, outfile: str) -> None:
+    """
+    CLI-tool for converting SPDX documents to Opossum documents.
+    To use, run: 'opossum_lib --infile <input file name> --out <output file path>'
```

Comment:
I would have expected the `opossum_lib --infile <input file name> --out <output file path>` to be auto-generated by click (but I might be wrong).
Having this hardcoded feels fragile: I somebody in the future renames any option, he might forget to adapt the hardcoded string.